src/appleseed/renderer/modeling/project/projectfilereader.cpp

```diff
@@ -1817,7 +1817,10 @@ namespace
             const XMLCh* const  chars,
             const XMLSize_t     length) override
         {
-            m_code = trim_both(transcode(chars));
+            if(!m_code.empty()){
```

Comment:
My theory is that with the fix you did in `write_osl_code()`, this change isn't needed. It shouldn't required really, and it looks a bit hacky (why would adding a space be always valid?). Could you test that theory?

Reply:
I already did. Even in the new setup the read stops at every special character (i.e. this function gets called for multiple blocks belonging to the same code).
The space is indeed hacky. I had a more explicit version at hand, which seemed a bit huge. Since "trim" only removes spaces, I'd count the amount of removed spaces and added them again as soon as a new codesnippet gets added.
I'll test your theory tomorrow again in different setups and report back.

Reply:
I would be quite surprised if with proper escaping the whole text wasn't read at once.